packages/one-app-bundler/webpack/loaders/provided-externals-loader.js

```diff
@@ -31,7 +31,9 @@ ${match[1]}.appConfig = Object.assign({}, ${match[1]}.appConfig, {
   },
 });
 
+if(global.getTenantRootModule === undefined || (global.getTenantRootModule && global.getTenantRootModule().holocron.name === '${options.moduleName}')){
```

Comment:
I'm wondering if we would need the second part of that statement. If the root module is the first module (script) added before the other modules, would the second statement ever be true? What I'm referring to specifically:
```
global.getTenantRootModule && global.getTenantRootModule().holocron.name === '${options.moduleName}'
```

Reply:
@Francois-Esquire would it possibly be if they are redeploying their root module with new externals?

Reply:
Yeah so this takes care of the following scenarios: 

1. When the root module is loaded for the first time, the global variable will be undefined so it sets it for the first time. (first part of the if statement)
2. If the root module gets updated and we have added new externals, it is no longer undefined so we go to the second part of the statement after the or || 
3. If the module that is being updated is the same module that set the variable originally (root module) then we allow it to override it. if not (another module incorrectly providing externals) then we skip and leave it as it is.

Reply:
This assumes the root module gets loaded into memory first every time which should be the case

Reply:
Ahh good point

Reply:
what if the package.json name does not match the holocron.name of the root module even though they are the same module.

Reply:
If that's the case and the root module updates externals, they won't be available until the server restarts 😬 ... I don't think there is a way to get the holocron module name from webpack at bundle time. Another alternative is to store yet another global variable saving the name of the module that set the externals initially and then compare it that way rather than comparing it to the holocron module name

Reply:
Another thing I noticed is with this approach is the following scenario: 

if you don't set any provideExternals on your root module, but you do on a child module. You will get the `warn: Module test-child-module attempted to provide externals. Only the root module can provide externals but it will still set the `global.getTenantRootModule()` (because it is undefined)

now if you want to remove providedExternals from the child module and add them correctly to the root module, it will require a server restart. Because the variable is already defined incorrectly by somebody else.

This is a very unlikely edge case, however I don't see any other way to fix it